src/scenic/core/regions.py

```diff
@@ -3049,7 +3049,14 @@ def __eq__(self, other):
 
     @cached
     def __hash__(self):
-        return hash((self.polygons, self.orientation, self.z))
+        return hash(
+            (
+                tuple(self._points) if self._points else None,
```

Comment:
This seems like an odd place to normalize the type. We don't want someone to pass a list of random points to `PolygonalRegion`, mutate that list, and have the mutation end up affecting the resulting sampled region. Probably better to convert to `tuple` in the initializer and use plain `self._points` here.

Reply:
Good call, I'll make the change.